lib/widgets/expandablelist_section.dart

```diff
@@ -0,0 +1,28 @@
+import 'package:flutter/material.dart';
+import 'package:mixeditems/repository/repo.dart';
+
+import '../models/listitem.dart';
+
+class ExpandableListSection implements ListItem {
+  final String title;
+  ExpandableListSection(this.title);
+
+  @override
+  Widget buildHead(BuildContext context) => const SizedBox.shrink();
+
+  @override
+  Widget buildBody(BuildContext context) {
+    return ExpansionTile(
+          title: Text(title),
+          controlAffinity: ListTileControlAffinity.leading,
+          children: createMockData(),
+    );
+  }
+
+  List<int> mockData = Repo().getMockData();
```

Comment:
This is now a public property of the ExpandableListSection and is initialised during construction.
It creates a new Repository, gets a snapshot of that data and saves it for the runtime in the property.
Usually you would not cache repository data this way but instead call the repository's method each time you need the data.
So it would make more sense to have the repository as a property and call it's method inside of the createMockData.

Reply:
I make "Repo" as a property on IconListSection and I put "getIconListItems" method into "createMockData" method.
initialize with new Repo().

And for the expandableList, I add "Repo"  as instance of constructor. And throw itself from Repo class.

Both look working well but which way is right..? or both are okay?

Reply:
Let's go for the constructor way to do it for now. Later we will introduce dependency injection :)

Reply:
Okay! I will do that.